python/ray/_private/event/export_event_logger.py

```diff
@@ -0,0 +1,137 @@
+import logging
+import pathlib
+import json
+import random
+import string
+import threading
+
+from typing import Union
+from datetime import datetime
+
+from ray.core.generated.export_event_pb2 import ExportEvent
+from ray.core.generated.export_submission_job_event_pb2 import (
+    ExportSubmissionJobEventData,
+)
+from ray._private.protobuf_compat import message_to_dict
+
+global_logger = logging.getLogger(__name__)
+
+# This contains the union of export event data types which emit events
+# using the python ExportEventLoggerAdapter
+ExportEventDataType = Union[ExportSubmissionJobEventData]
+
+
+def get_event_id():
+    return "".join([random.choice(string.hexdigits) for _ in range(18)])
+
+
+class ExportEventLoggerAdapter:
+    def __init__(self, source: ExportEvent.SourceType, logger: logging.Logger):
+        """Adapter for the Python logger that's used to emit export events."""
+        self.logger = logger
+        self.source = source
+
+    def send_event(self, event_data: ExportEventDataType):
+        # NOTE: Python logger is thread-safe,
+        # so we don't need to protect it using locks.
+        try:
+            event = self._create_export_event(event_data)
+        except TypeError:
+            global_logger.exception(
+                "Failed to create ExportEvent from event_data so no "
+                "event will be written to file."
+            )
+            return
+
+        event_as_str = self._export_event_to_string(event)
+
+        self.logger.info(event_as_str)
+        # Force flush so that we won't lose events
+        self.logger.handlers[0].flush()
```

Comment:
is this a pattern we follow in the other event exporters as well?

I think batching flushing is reasonable if we get significant performance improvements at the loss of some reliability

Reply:
since it is only used by a job, it is unlikely cause a problem, but I agree it is probably okay to lose some events here... (I don't think we guarantee flush every event after it is finished)

Reply:
Yeah, I added this behavior for consistency with the `EventLoggerAdapter` for existing python events https://github.com/ray-project/ray/blob/d8a85c560b46a82994f143cbf6cc430af0c5242f/python/ray/_private/event/event_logger.py#L105 The `LogEventReporter` for C++ events also flushes after each event by default https://github.com/ray-project/ray/blob/d8a85c560b46a82994f143cbf6cc430af0c5242f/src/ray/util/event.h#L103

Reply:
The export events from the python layer are unlikely to be created with very high volume (unlike tasks), so we can probably just keep this flush logic for now for consistency and optimize later if needed.